client/state/products-list/actions.js

```diff
@@ -1,35 +1,50 @@
-/**
- * External dependencies
- */
-import { mapValues } from 'lodash';
-
 /**
  * Internal dependencies
  */
 import wpcom from 'calypso/lib/wp';
-import { createProductObject } from './assembler';
 import {
 	PRODUCTS_LIST_RECEIVE,
 	PRODUCTS_LIST_REQUEST,
 	PRODUCTS_LIST_REQUEST_FAILURE,
 } from 'calypso/state/action-types';
+import { ensureNumericCost } from './assembler';
 
 import 'calypso/state/products-list/init';
 
 export function receiveProductsList( productsList ) {
+	// Since the request succeeded, productsList should be guaranteed non-null;
+	// thus, we don't have any safety checks before this line.
+
+	// Create a completely new products list to avoid mutating the original
+	const sanitizedProductsList = Object.entries( productsList ).reduce(
+		( acc, [ slug, product ] ) => {
+			acc[ slug ] = ensureNumericCost( product );
+			return acc;
+		},
+		{}
+	);
```

Comment:
This is another way how to write `mapValues`, in case you like it:
```js
Object.fromEntries(
  Object.entries( productsList ).map(
    ( [ slug, product ] ) => [ slug, ensureNumericCost( product ) ]
  )
);
```